crates/router/src/core/payment_methods.rs

```diff
@@ -1301,16 +1316,15 @@ pub async fn vault_payment_method(
 
     // throw back error if payment method is duplicated
     when(
-        Some(
-            db.find_payment_method_by_fingerprint_id(
-                &(state.into()),
-                key_store,
-                &fingerprint_id_from_vault,
-            )
-            .await
-            .change_context(errors::ApiErrorResponse::InternalServerError)
-            .attach_printable("Failed to find payment method by fingerprint_id")?,
+        db.find_payment_method_by_fingerprint_id(
+            &(state.into()),
+            key_store,
+            &fingerprint_id_from_vault,
         )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to find payment method by fingerprint_id")
+        .ok()
         .is_some(),
```

Comment:
1. We can call `is_ok()` instead of `ok().is_some()`.
2. We are ignoring the error here because of the `ok()`, can we log at least, with `inspect_err()`?